sdk/storage/azure-storage-queue/src/test/java/com/azure/storage/queue/spock/QueueAPITests.groovy

```diff
@@ -0,0 +1,485 @@
+import com.azure.storage.queue.models.StorageErrorException
+import org.junit.Assert
+import org.junit.Ignore
+import reactor.test.StepVerifier
+import spock.lang.Unroll
+
+import java.time.Duration
+import java.time.LocalDateTime
+import java.time.OffsetDateTime
+import java.time.ZoneOffset
+
+import static org.junit.Assert.assertEquals
+import static org.junit.Assert.assertFalse
+import static org.junit.Assert.assertNull
+
+class QueueAPITests extends APISpec {
+    def queueClient
+
+    static def testMetadata = Collections.singletonMap("metadata", "value")
+    static def createMetadata = Collections.singletonMap("metadata1", "value")
+
+    def setup() {
+        primaryQueueServiceClient = queueServiceBuilderHelper(interceptorManager).buildClient()
+        queueClient = primaryQueueServiceClient.getQueueClient(testResourceName.randomName("queue", 16))
+    }
+
+    def "Create queue with shared key from queue client"() {
+        when:
+        def createQueueResponse = queueClient.create()
+
+        then:
+        QueueTestHelper.assertResponseStatusCode(createQueueResponse, 201)
+    }
+
+    // TODO: Will implement the test after introduce the sas token generator
+    @Ignore
+    def "Create queue with sas token from queue client"() {
+
+    }
+
+    def "Delete exist queue from queue client"() {
+        given:
+        queueClient.create()
+        when:
+        def deleteQueueResponse = queueClient.delete()
+        QueueTestHelper.sleepInRecordMode(Duration.ofSeconds(30))
+        queueClient.enqueueMessage("This should fail")
+        then:
+        QueueTestHelper.assertResponseStatusCode(deleteQueueResponse, 204)
+        def e = thrown(StorageErrorException)
+        QueueTestHelper.assertExceptionStatusCodeAndMessage(e, 404, StorageErrorCode.QUEUE_NOT_FOUND)
+
+    }
+
+    def "Delete queue error from queue client"() {
+        when:
+        queueClient.delete()
+        then:
+        def e = thrown(StorageErrorException)
+        QueueTestHelper.assertExceptionStatusCodeAndMessage(e, 404, StorageErrorCode.QUEUE_NOT_FOUND)
+    }
+
+    def "Get properties from queue client"() {
+        given:
+        queueClient.create(testMetadata)
+        when:
+        def getPropertiesResponse = queueClient.getProperties()
+        then:
+        QueueTestHelper.assertResponseStatusCode(getPropertiesResponse, 200)
+        assertEquals(0, getPropertiesResponse.value().approximateMessagesCount())
+        assertEquals(testMetadata, getPropertiesResponse.value().metadata())
+    }
+
+    def "Get properties error from queue client"() {
+        when:
+        queueClient.getProperties()
+        then:
+        def e = thrown(StorageErrorException)
+        QueueTestHelper.assertExceptionStatusCodeAndMessage(e, 404, StorageErrorCode.QUEUE_NOT_FOUND)
+    }
+
+    @Unroll
+    def "Set and clear metadata from queue client"() {
+        given:
+        queueClient.create(matadataInCreate)
+        when:
```

Comment:
Could we add spacing between the test steps for readability?

Reply:
Do you mean space between given/when/then, etc?

Reply:
Yes, but this can be done later.

Reply:
Sure. Will address this later